guides/recipes/anti-patterns.md

```diff
@@ -0,0 +1,843 @@
+- Masking of issues that should trigger alerts
+- Lack of context for troubleshooting
+- Unpredictable system behavior under failure conditions
+
+### Example
+
+```elixir
+defmodule MyApp.Workers.UserSynchronizer do
+  use Oban.Worker, queue: :sync, max_attempts: 20
+
+  @impl Oban.Worker
+  def perform(%{args: %{"user_id" => user_id}}) do
+    # Poor error handling - will retry even for expected errors
+    case MyApp.Users.fetch(user_id) do
+      {:ok, user} ->
+        MyApp.ExternalService.sync_user(user)
+
+      {:error, _reason} ->
+        # Generic error, will trigger retry without context
+        raise "Failed to sync user #{user_id}"
+    end
+  end
+end
+```
+
+### Refactoring
+
+Implement thoughtful error handling with appropriate retry behavior:
+
+```elixir
+defmodule MyApp.Workers.UserSynchronizer do
+  use Oban.Worker, queue: :sync, max_attempts: 5
+  require Logger
+
+  @impl Oban.Worker
+  def perform(%{args: %{"user_id" => user_id} = args, attempt: attempt}) do
+    case MyApp.Users.fetch(user_id) do
+      {:ok, user} ->
+        case MyApp.ExternalService.sync_user(user) do
+          {:ok, _} ->
+            :ok
+
+          {:error, %{status: 400, reason: reason}} ->
+            # Client error - don't retry
+            Logger.warn("User sync failed with client error: #{reason}", args: args)
+            {:discard, :client_error}
+
+          {:error, %{status: status}} when status >= 500 ->
+            # Server error - worth retrying with backoff
+            backoff = :math.pow(2, attempt) |> round()
+            Logger.warn("User sync failed with server error: #{status}", args: args)
+            {:snooze, backoff}
+
+          {:error, :timeout} ->
+            # Network issue - worth retrying
+            Logger.warn("User sync timed out", args: args)
+            {:error, :timeout}
+        end
+
+      {:error, :not_found} ->
+        # Don't retry when the user doesn't exist
+        Logger.info("Cannot sync user #{user_id} - not found", args: args)
+        {:discard, :user_not_found}
+
+      {:error, reason} ->
+        # Database error - worth retrying
+        Logger.error("Failed to fetch user #{user_id}: #{inspect(reason)}", args: args)
+        {:error, reason}
+    end
+  end
+end
+```
+
+### Additional Remarks
+
+| Return Value | When to Use |
+|--------------|-------------|
+| `:ok` or `{:ok, result}` | Job completed successfully |
+| `{:error, reason}` | Transient error, should be retried |
+| `{:discard, reason}` | Expected failure, don't retry |
+| `{:snooze, seconds}` | Temporary issue, retry after delay |
+| `{:cancel, reason}` | Cancel this job and dependent jobs |
+
+Additional recommendations:
+
+- ✅ Log context-rich information for all errors
+- ✅ Use exponential backoff for retries
+- ✅ Set appropriate `max_attempts` values based on error types
+- ✅ Consider uniqueness constraints for idempotency
+- ✅ Use structured error returns rather than exceptions
+
+## Best Practices for Oban Usage
```

Comment:
This whole section was the AI being too eager, I left it here in case you find it valuable. Though it should be in another guide and not in the anti-patterns one